### ISSUE TITLE
[Feature Request] Hot reloading TLS certificates in FRPS 

### ISSUE THREAD

**Issue by splinter98** [fatedier/frp#2946]
### Describe the feature request

Currently in FRPS TLS certificates are only loaded during the initialisation of the server, so if the signed certificate expires or are replaced FRPS needs to be restarted which will break any client connections.

Looking  at how other programs handle this there seems to be two main approaches.

1. Catch any SIGHUP signals sent to the FRPS process and reload the certificate
2. Watch the files and reload the files when they changed

The former seemed simpler to implement so I have a working version at: [splinter98/frp](https://github.com/splinter98/frp/tree/ssh-renewal) on the ssh-renewal branch.

### Describe alternatives you've considered

I've also considered looking implementing the file monitoring approach but this would introduce another 3rd party library dependency.

### Affected area

- [ ] Docs
- [ ] Installation
- [ ] Performance and Scalability
- [X] Security
- [X] User Experience
- [ ] Test and Release
- [ ] Developer Infrastructure
- [ ] Client Plugin
- [X] Server Plugin
- [ ] Extensions
- [ ] Others

**Comment by keliansb** [comment]
Would love to see this feature implemented!

**Comment by srekkas** [comment]
Hi. 

We need it too. I am testing if frps/c reloads certificates right now.

**Comment by srekkas** [comment]
Yep, it wont reload certificates. We use cert-manager to generate certificates and some automatic reload is must for both frps and frpc.

We can implement sidecar for frpc, which watch certificates and query config reload api, if it will reload certificates.
But what about frps, i do not want use external watcher and reload frps pod, additional config, updates etc...

**Comment by srekkas** [comment]
Hi, can it be done :)
